lib/seahorn/BvOpSem2.cc

```diff
@@ -914,12 +940,19 @@ class OpSemMemManager {
     for (unsigned i = 0; i < len; i += sem_word_sz) {
       Expr offset = bv::bvnum(i, ptrSzInBits(), m_efac);
       Expr dIdx = i == 0 ? dPtr : mk<BADD>(dPtr, offset);
-      unsigned word = 0;
+      // copy bytes from buffer to word - word must accommodate largest supported word size
+      unsigned long word = 0;  // 8 bytes because assumed largest supported sem_word_sz = 8
       for (int byte = sem_word_sz - 1; byte >= 0; byte--) {
-        word = word << 8;
+        word = word << 8L;
         if (i + byte < len)
           word += sPtr[i + byte];
       }
+      if (sem_word_sz == 8) {
+        // special case: for word size of 8, swap low 32 bits with high 32 bits
+        // necessary because earlier addresses should be in left half of word
+        //   eg. 32-bit: 0x01234567 0x89abcdef -> 64-bit: 0x89abcdef01234567 -> flipped: 0x0123456789abcdef
```

Comment:
I still don't get this. Would be good to meet to talk about this in person. 
Still, the code should be factored out into special functions so that the use (here) is clearer, and comments (on the function) make sense independently of the use.